extension/src/connect/index.ts

```diff
@@ -0,0 +1,118 @@
+import { commands, ExtensionContext, SecretStorage } from 'vscode'
+import { validateTokenInput } from './inputBox'
+import { STUDIO_ACCESS_TOKEN_KEY, isStudioAccessToken } from './token'
+import { STUDIO_URL } from './webview/contract'
+import { Resource } from '../resourceLocator'
+import { ViewKey } from '../webview/constants'
+import { MessageFromWebview, MessageFromWebviewType } from '../webview/contract'
+import { BaseRepository } from '../webview/repository'
+import { Logger } from '../common/logger'
+import { getInput, getValidInput } from '../vscode/inputBox'
+import { Title } from '../vscode/title'
+import { openUrl } from '../vscode/external'
+import { ContextKey, setContextValue } from '../vscode/context'
+import { RegisteredCommands } from '../commands/external'
+
+export class Connect extends BaseRepository<undefined> {
+  public readonly viewKey = ViewKey.CONNECT
+
+  private readonly secrets: SecretStorage
```

Comment:
[F] This seems like a better API to use instead of saving the token directly into the User's config.